oci/archive/reader.go

```diff
@@ -0,0 +1,93 @@
+package archive
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/containers/image/v5/internal/tmpdir"
+	"github.com/containers/image/v5/oci/layout"
+	"github.com/containers/image/v5/types"
+	"github.com/containers/storage/pkg/archive"
+	imgspecv1 "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/pkg/errors"
+)
+
+type reader struct {
+	manifest *imgspecv1.Index
+	tempDirOCIRef
+}
+
+func CreateUntarTempDirReader(src string, sys *types.SystemContext) (*reader, error) {
+	// TODO: This can take quite some time, and should ideally be cancellable using a context.Context.
```

Comment:
Note sure what "this" refers to and how we could cancel. But if we want to do add cancellation we should change the signature of the function and add a `context.Context`. This way we don't break the API in the future.

Reply:
Will look into add the `context.Context`.  I grab the same code from oci_transport.go to untar the archive. The comments are from https://github.com/containers/image/blob/f1a6f9b50d0286bf208935d4b9ab48117294ac31/oci/archive/oci_transport.go#L185

Reply:
(Drive-by comment, I haven’t read the full PR.) These TODO comments are sprinkled around the code base from the time we’ve added `context.Context` to the transports, but we didn’t actually implement cancellation in many places because `io.Copy` and local file operations are not trivially cancellable in Go.

So it would be, I guess, consistent, to leave that s a TODO instead of implementing — but I do agree that worrying about making it eventually _possible_ through the API by at least passing (and ignoring) a `context.Context` makes sense.